src/utils.ts

```diff
@@ -43,37 +43,46 @@ export const runIframe = (
 ) => {
   return new Promise<AuthenticationResult>((res, rej) => {
     const iframe = window.document.createElement('iframe');
+
     iframe.setAttribute('width', '0');
     iframe.setAttribute('height', '0');
     iframe.style.display = 'none';
 
     const removeIframe = () => {
       if (window.document.body.contains(iframe)) {
         window.document.body.removeChild(iframe);
+        window.removeEventListener('message', iframeEventHandler, false);
       }
     };
 
+    let iframeEventHandler: (e: MessageEvent) => void;
+
     const timeoutSetTimeoutId = setTimeout(() => {
       rej(new TimeoutError());
       removeIframe();
     }, timeoutInSeconds * 1000);
 
-    const iframeEventHandler = function (e: MessageEvent) {
+    iframeEventHandler = function (e: MessageEvent) {
       if (e.origin != eventOrigin) return;
       if (!e.data || e.data.type !== 'authorization_response') return;
+
       const eventSource = e.source;
+
       if (eventSource) {
         (eventSource as any).close();
       }
+
       e.data.response.error
         ? rej(GenericError.fromPayload(e.data.response))
         : res(e.data.response);
+
       clearTimeout(timeoutSetTimeoutId);
-      window.removeEventListener('message', iframeEventHandler, false);
```

Comment:
I think the danger of removing this one would be that, if you did 2 successful `getTokenSilently({ ignoreCache: true })` within `CLEANUP_IFRAME_TIMEOUT_IN_SECONDS` seconds of each other, the first `iframeEventHandler` would still be around and handle the second `getTokenSilently`'s `postMessage`

Reply:
Yes, good catch - I'll reinstate it.